### PR TITLE
Added a disconnect event to the MutationObserver listener

### PR DESCRIPTION
**Pull request by pankaj-arunsingh**

### CODE REVIEW

**Comment by pankaj-arunsingh** [comment]
Modified for showing scrollbar when element become visible after hidden and added debounce for mutation events.

File: src/simple-scrollbar.js [hunk 1]
```diff
@@ -37,7 +37,36 @@
       d.removeEventListener('mouseup', stop);
     }
   }
-
+  function getHiddenParent(element) {
+    var hiddenParent = "";
```
Comment (ywonglanecove) [inline src/simple-scrollbar.js]: hidden should not be a string

Comment (pankaj-arunsingh) [inline src/simple-scrollbar.js]: then shall I leave the variable unassigned?

File: src/simple-scrollbar.js [hunk 2]
```diff
@@ -68,31 +97,44 @@
...
-
-    var mutationObserver = new MutationObserver(function(mutations){
+    if (el.offsetHeight === 0) {
+      var visibilityObserver = new MutationObserver(function (visibilityMutations) {
         moveBar();
+        visibilityObserver.disconnect();
+      });
+      var config = { attributes: true, childList: false, characterData: false, subtree: false, attributeFilter: ["class", "style"] };
+      var hiddenParent = el;
+      if (hiddenParent.style.display !== "none") {
```
Comment (ywongau) [inline src/simple-scrollbar.js]: checking style.display is incorrect, it fails if it has a css class that sets display:none.
check offsetParent !== null instead

File: src/simple-scrollbar.js (hunk 1 above)
Comment (ywongau) [inline src/simple-scrollbar.js]: hiddenParent should not be a string

File: src/simple-scrollbar.js [hunk 3]
```diff
@@ -37,7 +37,36 @@
       d.removeEventListener('mouseup', stop);
     }
   }
-
+  function getHiddenParent(element) {
+    var hiddenParent = "";
+    while (element.parentNode !== null && element.parentNode.offsetHeight === 0) {
+      if (element.parentNode.style.display === "none") {
```
Comment (ywongau) [inline src/simple-scrollbar.js]: this is incorrect. if u add a new test that applies a class to body to make it invisible, the test will fail
if u add another test that set scrollbar's viewport to display:none and then set viewport's container to display:none, it will also fail

File: src/simple-scrollbar.js [hunk 4]
```diff
@@ -37,7 +37,36 @@
       d.removeEventListener('mouseup', stop);
     }
   }
-
+  function getHiddenParent(element) {
+    var hiddenParent = "";
+    while (element.parentNode !== null && element.parentNode.offsetHeight === 0) {
```
Comment (ywongau) [inline src/simple-scrollbar.js]: use offsetParent === null to test if an element is hidden

**Review by ywongau — CHANGES_REQUESTED** [review]

**Comment by pankaj-arunsingh** [comment]
Committed the review comment changes made by you. :-)
Thanks
